### ISSUE TITLE
FP redundant_static_lifetimes

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9890]
### Summary

.

### Lint Name

redundant_static_lifetimes

### Reproducer

I tried this code:

```rust
static mut TEST: &'static mut [isize] = &mut [1];
static mut EMPTY: &'static mut [isize] = &mut [];

pub fn main() {
    unsafe {
        TEST[0] += 1;
        assert_eq!(TEST[0], 2);
    }
}
```

I saw this happen:
`cargo clippy --fix -- -Aclippy::all -Wclippy::redundant_static_lifetimes`
```
    Checking clpy v0.1.0 (/tmp/clpy)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0594]: cannot assign to `TEST[_]`, which is behind a `&` reference
 --> src/main.rs:6:9
  |
6 |         TEST[0] += 1;
  |         ^^^^^^^^^^^^ cannot assign

error: aborting due to previous error

For more information about this error, try `rustc --explain E0594`.
Original diagnostics will follow.

warning: statics have by default a `'static` lifetime
 --> src/main.rs:1:19
  |
1 | static mut TEST: &'static mut [isize] = &mut [1];
  |                  -^^^^^^^------------ help: consider removing `'static`: `&[isize]`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_static_lifetimes
  = note: requested on the command line with `-W clippy::redundant-static-lifetimes`

warning: statics have by default a `'static` lifetime
 --> src/main.rs:2:20
  |
2 | static mut EMPTY: &'static mut [isize] = &mut [];
  |                   -^^^^^^^------------ help: consider removing `'static`: `&[isize]`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_static_lifetimes

warning: static `EMPTY` is never used
 --> src/main.rs:2:12
  |
2 | static mut EMPTY: &'static mut [isize] = &mut [];
  |            ^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: `clpy` (bin "clpy") generated 3 warnings (run `cargo fix --bin "clpy"` to apply 2 suggestions)
warning: `clpy` (bin "clpy" test) generated 3 warnings (3 duplicates)
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
```

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
